src/components/ArticlePage.jsx

```diff
@@ -1,94 +1,93 @@
-import React, { Component } from "react";
-import CreateArticle from "./CreateArticle";
-import { Button } from "semantic-ui-react";
-import { connect } from "react-redux";
+import React, { useState, useEffect } from "react";
 import axios from "axios";
 import ArticleContent from "./ArticleContent"
+import { useSelector } from "react-redux";
 
-class ArticlePage extends Component {
-  state = {
-    renderForm: false,
-    articles: [],
-    singleArticle: null,
-  };
+const ArticlePage = () => {
+  const [articles, setArticles] = useState([])
+  const [singleArticle, setSingleArticle] = useState(null)
+  const [message, setMessage] = useState(null)
+  const userRole = useSelector(state => state.currentUser.role)
 
-  componentDidMount = () => {
-    this.getArticles();
-  };
 
-  getArticles = async () => {
+  useEffect(() => {
+    getArticles()
+  }, [])
+
+  const getArticles = async () => {
     let response;
-    response = await axios.get("/articles")
-    this.setState({ articles: response.data.articles });
+    if (userRole === 'editor') {
+      response = await axios.get("/admin/articles", {
+        params: {
+          published: false
+        }
+      })
+    } else {
+      response = await axios.get("/articles")
+    }
+    setArticles(response.data.articles);
   };
 
-  getSingleArticle = async (event) => {
+  const getSingleArticle = async (event) => {
     let id = event.target.parentElement.dataset.id;
     let response = await axios.get(`/articles/${id}`);
```

Comment:
What are we expecting to see here? Should the request be made at /admin/articles?

Reply:
There is no need to make a request to the admin controller for a single article